### ISSUE TITLE
Indeterminate check box state is not clearly identifiable in high contrast themes

### ISSUE THREAD

**Issue by StephenLPeters** [microsoft/microsoft-ui-xaml#2592]
Copied from internal [issue](https://microsoft.visualstudio.com/OS/_workitems/edit/26425988?src=WorkItemMention&src-action=artifact_link).
Expected:
Indeterminate check box state should be as below in High contrast themes:
HC Black: Check box should be in 'Same as the disabled but change the green to blue.' 
HC White: Check box should be in 'Same as the disabled but change the maroon to purple.' 

Actual:
Indeterminate check box state appears as below in High contrast themes:
HC Black: Check box is in 'White Color with white square shaped fill in the middle' 
HC White: Check box is in 'Black Color with black square shaped fill in the middle' 

![inderminate-HC black](https://user-images.githubusercontent.com/7758786/83689367-c44f3900-a5a3-11ea-94a9-35fd3190086c.png)
![inderminate-HC white](https://user-images.githubusercontent.com/7758786/83689370-c618fc80-a5a3-11ea-97a2-d14785877e5c.png)
![inderminate-Normal](https://user-images.githubusercontent.com/7758786/83689372-c74a2980-a5a3-11ea-893c-bf8915ea7d81.png)

**Comment by Felix-Dev** [comment]
Should the checkbox look with PointerOver in indeterminate state be changed as well? And if yes, how should it then look like in HC Black and HC White?

In rest mode with requested change applied:
![image](https://user-images.githubusercontent.com/1398851/84091527-ece89000-a9f4-11ea-89cf-9166e25776ee.png)

Unchanged (as of yet) pointer over look:
![image](https://user-images.githubusercontent.com/1398851/84091538-fd990600-a9f4-11ea-9b0c-aa1d26f536b9.png)

The square-shaped fill in the middle could be changed to black (purble in HC White) as a possible suggestion. This would follow the current design of the checkbox when it's in checked state (IsChecked=true).

**Comment by YuliKl** [comment]
This issue has been resolved with recent style updates to CheckBox control.
